core/utils_test.go

```diff
@@ -0,0 +1,133 @@
+package core_test
```

Comment:
Why did you choose `core_test` instead of `core` for this file?
Since you want to test a internal function `runUntilComplete`, isn't it the best to simply use `core`?